### PR TITLE
support vertexai featureonlinestore_featureview GA

### PR DESCRIPTION
**Pull request by kkram01**
This PR is to support vertexai [FeatureOnlineStore.Featureview](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.featureOnlineStores.featureViews#resource:-featureview) Resource with Terraform GA provider. 

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_vertex_ai_feature_onlinestore_featureview
```

### CODE REVIEW

**Review by slevenick — CHANGES_REQUESTED** [review]
Failing with:
        An argument named "force_destroy" is not expected here.

**Comment by kkram01** [comment]
> Failing with: An argument named "force_destroy" is not expected here.

Looks like the current version of featureonlinestore doesn't support force_destroy? I've a PR in for adding that and some beta fields to it. removed that field in the example for now.

File: mmv1/templates/terraform/encoders/vertex_ai_featureonlinestore_featureview.go.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+if v, ok := d.GetOk("feature_online_store"); ok {
+	re := regexp.MustCompile("projects/(.+)/locations/(.+)/featureOnlineStores/(.+)$")
+	if parts := re.FindStringSubmatch(v.(string)); parts != nil {
+		d.Set("region", parts[2])
```
Comment (slevenick) [inline mmv1/templates/terraform/encoders/vertex_ai_featureonlinestore_featureview.go.erb:4]: Again, why region?

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/templates/terraform/custom_import/vertex_ai_featureonlinestore_featureview.go.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+if err != nil {
+	return nil, fmt.Errorf("Error constructing id: %s", err)
+}
+d.SetId(id)
+
+featureOnlineStore := d.Get("feature_online_store").(string)
+
+re := regexp.MustCompile("^projects/(.+)/locations/(.+)/featureOnlineStores/(.+)$")
+if parts := re.FindStringSubmatch(featureOnlineStore); parts != nil {
+	d.Set("region", parts[2])
```
[the remark this answers is not in the document]
Comment (shotarok) [reply mmv1/templates/terraform/custom_import/vertex_ai_featureonlinestore_featureview.go.erb:19]: Sorry for suddenly jumping into this PR. The VertexAI URL uses region in the host name.  

https://github.com/GoogleCloudPlatform/magic-modules/blob/4d160cb257e40ad9228a94766acfa3b77becb14e/mmv1/products/vertexai/product.yaml#L17-L23

Therefore, `google_vertex_ai_endpoint` started using `location`, but `region` was added later in my understanding. ref. https://github.com/hashicorp/terraform-provider-google/issues/16478

**Comment by kkram01** [comment]
@slevenick as discuused, raised this PR (https://github.com/GoogleCloudPlatform/magic-modules/pull/9696) as an alternate approach with the standard_url structure. pls review and share your thoughts on which would be the best way forward for this. Thanks!

**Comment by slevenick** [comment]
> @slevenick as discuused, raised this PR (#9696) as an alternate approach with the standard_url structure. pls review and share your thoughts on which would be the best way forward for this. Thanks!

I think we should proceed with the alternate approach.

**Comment by kkram01** [comment]
> > @slevenick as discuused, raised this PR (#9696) as an alternate approach with the standard_url structure. pls review and share your thoughts on which would be the best way forward for this. Thanks!
> 
> I think we should proceed with the alternate approach.

Sure, Could you pls review that PR as well since you have the context related to this resource?

**Comment by melinath** [comment]
Closing this PR per discussion above

**Pull request closed without merging**